apps/local_laplacian/local_laplacian_generator.cpp

```diff
@@ -82,20 +76,19 @@ class LocalLaplacian : public Halide::Generator<LocalLaplacian> {
         float eps = 0.01f;
         color(x, y, c) = outGPyramid[0](x, y) * (floating(x, y, c)+eps) / (gray(x, y)+eps);
 
-        Func output("local_laplacian");
         // Convert back to 16-bit
         output(x, y, c) = cast<uint16_t>(clamp(color(x, y, c), 0.0f, 1.0f) * 65535.0f);
+    }
 
-
-
+    void schedule() {
         /* THE SCHEDULE */
         remap.compute_root();
 
         if (get_target().has_gpu_feature()) {
             // gpu schedule
             Var xi, yi;
-            output.compute_root().gpu_tile(x, y, xi, yi, 16, 8);
-            for (int j = 0; j < J; j++) {
+            Func(output).compute_root().gpu_tile(x, y, xi, yi, 16, 8);
```

Comment:
The only alternative I can think of is reorganizing Func, Stage, and Output<> to share a common base class that has the methods. Func/Output has more methods than Stage though. Also the base class would probably have to use CTRP to return the correct type from the scheduling calls.